pya/aserver.py

```diff
@@ -225,93 +376,54 @@ def play(self, asig, onset: Union[int, float] = 0, out: int = 0, **kwargs):
         out: int
             Output channel
         """
+        if not self.is_active:
+            raise RuntimeError("Aserver not active")
         self._stop = False
 
-        sigid = id(asig)  # for copy check
         if asig.sr != self.sr:
             asig = asig.resample(self.sr)
+
         if onset < 1e6:
             rt_onset = time.time() + onset
         else:
             rt_onset = onset
-        idx = np.searchsorted(self.srv_onsets, rt_onset)
-        self.srv_onsets.insert(idx, rt_onset)
+
         if asig.sig.dtype != self.backend.dtype:
-            warn("Not the same type. ")
-            if id(asig) == sigid:
-                asig = copy.copy(asig)
-            asig.sig = asig.sig.astype(self.backend.dtype)
-        # copy only relevant channels...
-        nchn = min(asig.channels, self.channels - out)  # max number of copyable channels
-        # in: [:nchn] out: [out:out+nchn]
-        if id(asig) == sigid:
-            asig = copy.copy(asig)
-        if len(asig.sig.shape) == 1:
-            asig.sig = asig.sig.reshape(asig.samples, 1)
-        asig.sig = asig.sig[:, :nchn].reshape(asig.samples, nchn)
-        # asig.channels = nchn
-        # so now in callback safely copy to out:out+asig.sig.shape[1]
-        self.srv_asigs.insert(idx, asig)
-        self.srv_curpos.insert(idx, 0)
-        self.srv_outs.insert(idx, out)
-        if 'block' in kwargs and kwargs['block']:
+            asig_data = asig.sig.astype(self.backend.dtype)
+        else:
+            asig_data = asig.sig
```

Comment:
before I had `copy.copy(asig)` as otherwise asig_data would be merely a reference to the original data. If that asig would be changed (outside) it could get lost. I assume the risk is negligible but `asig_data[:] = asig.sig` would be safe (but needed only if (i) data is not changed, i.e. read-only, (ii) there is no risk that the asig used when calling play() is subjected to changes until the scheduled event will be played... imo copying is safer, but costs time and memory...